gammapy/catalog/fermi.py

```diff
@@ -47,52 +48,469 @@ def compute_flux_points_ul(quantity, quantity_errp):
     return 2 * quantity_errp + quantity
 
 
-class SourceCatalogObject4FGL(SourceCatalogObject):
+class SourceCatalogObjectFermiBase(SourceCatalogObject):
+    """Base class for Fermi-LAT catalogs."""
+
+    asso = ["ASSOC1", "ASSOC2", "ASSOC_TEV", "ASSOC_GAM1", "ASSOC_GAM2", "ASSOC_GAM3"]
+
+    def __str__(self):
+        return self.info()
+
+    def info(self, info="all"):
+        """Summary info string.
+
+        Parameters
+        ----------
+        info : {'all', 'basic', 'more', 'position', 'spectral','lightcurve'}
+            Comma separated list of options
+        """
+        if info == "all":
+            info = "basic,more,position,spectral,lightcurve"
+
+        ss = ""
+        ops = info.split(",")
+        if "basic" in ops:
+            ss += self._info_basic()
+        if "more" in ops:
+            ss += self._info_more()
+        if "position" in ops:
+            ss += self._info_position()
+            if not self.is_pointlike:
+                ss += self._info_morphology()
+        if "spectral" in ops:
+            ss += self._info_spectral_fit()
+            ss += self._info_spectral_points()
+        if "lightcurve" in ops:
+            ss += self._info_lightcurve()
+        return ss
+
+    def _info_basic(self):
+        """Print basic info."""
+        d = self.data
+        keys = self.asso
+        ss = "\n*** Basic info ***\n\n"
+        ss += "Catalog row index (zero-based) : {}\n".format(d["catalog_row_index"])
+        ss += "{:<20s} : {}\n".format("Source name", d["Source_Name"])
+        ss += "{:<20s} : {}\n".format("Extended name", d["Extended_Source_Name"])
+
+        def get_nonentry_keys(keys):
+            vals = [d[_].strip() for _ in keys]
+            return ", ".join([_ for _ in vals if _ != ""])
+
+        associations = get_nonentry_keys(keys)
+        ss += "{:<16s} : {}\n".format("Associations", associations)
+        try:
+            ss += "{:<16s} : {:.3f}\n".format("ASSOC_PROB_BAY", d["ASSOC_PROB_BAY"])
+            ss += "{:<16s} : {:.3f}\n".format("ASSOC_PROB_LR", d["ASSOC_PROB_LR"])
+        except (KeyError):
+            pass
+        try:
+            ss += "{:<16s} : {}\n".format("Class1", d["CLASS1"])
+        except (KeyError):
+            ss += "{:<16s} : {}\n".format("Class", d["CLASS"])
+        try:
+            ss += "{:<16s} : {}\n".format("Class2", d["CLASS2"])
+        except (KeyError):
+            pass
+
+        tevcat_flag = d["TEVCAT_FLAG"]
+        if tevcat_flag == "N":
+            tevcat_message = "No TeV association"
+        elif tevcat_flag == "P":
+            tevcat_message = "Small TeV source"
+        elif tevcat_flag == "E":
+            tevcat_message = "Extended TeV source (diameter > 40 arcmins)"
+        else:
+            tevcat_message = "N/A"
+        ss += "{:<16s} : {}\n".format("TeVCat flag", tevcat_message)
+        return ss
+
+    @abc.abstractmethod
+    def _info_more(self):
+        return "\n"
```

Comment:
Please put `pass` for abstract methods, not some return value like "\n" that's never used.